dm/pkg/etcdutil/etcdutil_test.go

```diff
@@ -20,8 +20,10 @@ import (
 	"testing"
 	"time"
 
+	"github.com/phayes/freeport"
```

Comment:
we have no evidence that the failed CI is caused by port conflict, and I guess in fact that is not the root cause because port conflict will fail the syscall immediately rather than blocking.
And your PR is not a solution because it can't handle time-of-check to time-of-use

Reply:
yes, port conflicts is not the root cause root cause is updated in PR description

but i think this is no harm and more intuitiveto use free port in test ?

Reply:
this is already a free port

Reply:
we not have port confilcts problem in current codebase, but it have a chance that if someone start another etcd server with **hardcode** port 8361 in other test case, what's more, some other test with embed etcd also use a free port in config such as https://github.com/pingcap/tiflow/blob/afa9a2bf57e0cbff874af06748a9b71d63eef6ca/dm/dm/worker/server_test.go#L105 and https://github.com/pingcap/tiflow/blob/afa9a2bf57e0cbff874af06748a9b71d63eef6ca/dm/pkg/election/election_test.go#L60

so i think we better not keep this hardcode port ?

Reply:
Yes, we can reduce conflict caused by random port of other tests. I prefer we use `tempurl` rather than import a new library

Reply:
Sounds reasonable, replace to `tempurl` in https://github.com/pingcap/tiflow/pull/4175/commits/6ea37fa20e76f2b79018cc4a10b6b067749bc6c9